File: reacjilator-java-jslack/src/main/java/slackapp_backend/service/LangCodes.java
package slackapp_backend.service;

import com.amazonaws.util.IOUtils;
import com.google.gson.GsonBuilder;

import java.io.IOException;
import java.io.InputStream;
import java.util.Map;
import java.util.Set;

public class LangCodes {

    private static final String FILENAME = "langcode.json";

    private final Map<String, String> allData;
    private final Set<String> allKeys;

    public LangCodes() throws IOException {
        ClassLoader classLoader = LangCodes.class.getClassLoader();
        try (InputStream fileResource = classLoader.getResourceAsStream(FILENAME)) {
            String jsonStr = IOUtils.toString(fileResource);
            this.allData = new GsonBuilder().create().fromJson(jsonStr, Map.class);
            this.allKeys = this.allData.keySet();
        }
    }

    public Map<String, String> getAllData() {
        return allData;
    }

    public Set<String> getAllKeys() {
        return allKeys;
    }

}

File: reacjilator-scala-jslack/src/main/scala/slackapp/service/LangCodes.scala
package slackapp.service

import com.amazonaws.util.IOUtils
import com.google.gson.GsonBuilder
import slackapp.util.LoanPattern.using

import scala.collection.JavaConverters._

class LangCodes {
  import LangCodes._

  lazy val allData: Map[String, String] = {
    val classLoader = classOf[LangCodes].getClassLoader
    using(classLoader.getResourceAsStream(filename)) { resource =>
      val jsonStr = IOUtils.toString(resource)
      val gson    = new GsonBuilder().create

      gson.fromJson(jsonStr, classOf[java.util.Map[_, _]]).asScala.map { case (k, v) => k.toString -> v.toString }.toMap
    }
  }
  lazy val allKeys = allData.keys.toSeq

}

object LangCodes {

  private val filename = "langcode.json"

}
